samples/nrf9160/https_client/src/main.c

```diff
@@ -169,18 +170,24 @@ void main(void)
 	}
 	printk("OK\n");
 
-	err = getaddrinfo(HTTPS_HOSTNAME, NULL, &hints, &res);
+	/* Allow time for dual stack LTE to assign IPv6 before calling getaddrinfo() */
+	k_msleep(DUAL_STACK_DELAY_MS);
```

Comment:
I do not like this solution because it is just an arbitrary delay, which in some cases is also unnecessary.
The correct way to handle this would be to have a sample-level Kconfig to select which IP version to use and switch on that. In case of IPv6 we need to show how to handle the connectivity notification properly, either as done in `pdn.c.` using the AT notification directly, or by inclusion of the `pdn` library.

Reply:
Note that in the general case (you don't know in advance if the carrier supports dual stack, or if the address is dual stack), you should allow an opportunity for the "ipv6" event so that you can follow RFC6724.

This is my plan for adding proper PDN event handling -- in the general case wait for the "ipv6" event, and proceed as soon as it occurs, but with a (configurable) timeout to proceed with IPv4 only.  For example the modem may request dual stack, and the target address is dual stack, but if the particular carrier being used does not (maybe the device is roaming to a different carrier than normal), then you don't want to wait forever for an IPv6 address that you normally would get.

In the case where the user has explicitly configured the PDN as IPV4 only then we can skip the check.

The case where the PDN is still dual stack, but you pass a hint to only use the IPv4 address (the original code) is a bit unusual in my opinion, because it breaks the purpose of DNS, which is to abstract away the technical address by using the DNS name. I can't think of any scenario where you would want to do that.

If you have more detail on what is meant by "change the default from IPv4 to IPv6", then let me know. (Because my understanding is that the technical default is "IPv6 if you can, otherwise IPv4", and the only reason the previous example was IPv4 was because someone deliberately overrode that by passing the IPv4 only hint.)

Reply:
I did find the reference to the default PDN being IPV4V6:  https://github.com/nrfconnect/sdk-nrf/blob/main/lib/pdn/Kconfig#L42

It is also the default with LwM2M: https://github.com/nrfconnect/sdk-nrf/blob/main/lib/bin/lwm2m_carrier/Kconfig#L141

I also did find another sample that used the same hack (for a sample) of simply adding a delay: https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf9160/gnss/src/main.c#L216

But, as mentioned in my first reply, I will add some code to handle PDN events properly. (Where properly = wait for "ipv6" and continue as soon as it is available, with a timeout for when the system is configured for dual stack but IPv6 is not available, e.g. not provided by the carrier; and with an override for when the system has been deliberately changed to IPv4 only).

Reply:
Okay. I have updated the code to wait for the PDN to be fully activated (for the existing default dual stack configuration).

If the PDN defaults are overridden to IPv4 only, then the wait is skipped (it would always time out). I still have the example overlay that does this override.

I updated the examples to keep the scenario where the carrier does not support IPv4 as the first example (if this is what you meant by "don't change the default from IPv4 to IPv6".

The second example then shows the scenario where the carrier does support IPv6. (Both scenario 1 and 2 are using the default, dual-stack, configuration).

I added a third example to show when the overlay is used, as the output is now different (when overridden to IPv4 only, it skips the check/timeout for IPv6).